CRM/Report/Form.php

```diff
@@ -4893,4 +4893,26 @@ public function joinGroupTempTable($baseTable, $field, $tableAlias) {
     }
   }
 
+  /**
+   * Get all labels for fields that are used in a group concat.
+   *
+   * @param string $options
+   *   comma separated option values.
+   * @param array $group
+   *   array of option value/label pairs.
+   *
+   * return string
+   */
+  public function getGroupLabels($options, $group) {
```

Comment:
so, this doesn't need to be used only for group_concat, so perhaps just call it labels, ie. it would work with a field with no comma in it. I would go a step further and pass a bit more in eg 

public function getLabels($options, $baoName, $fieldName) {
            $values = $types = explode(',', $options);
            foreach ($values as $value) {
              $labels[] = CRM_Core_PseudoConstant::getLabel($baoName,, $fieldName, $value);
            }
          // implode labels etc
}

I would say CRM_Core_PseudoConstant::getLabel($baoName,, $fieldName, $value); is the best function to call from alterDisplay when not multiple, but for simplicity all rows could just call this one - e.g

$row['contribution_status_id'] => $this->getLabels('CRM_Contribute_BAO_Contribution', 'contribution_status_id', $value);

Extra aside - that code above is lifted from BAO_Contact_Query::convertToPseudoNames() but handles custom separators too. Perhaps getLabels should live on the pseudoconstant class & be called from more places.